### PR TITLE
feature/CWALL-155

### PR DESCRIPTION
**Pull request by maikel-maas**
Migration files #8 are in CWALL-153

### CODE REVIEW

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts [hunk 1]
```diff
@@ -85,6 +85,12 @@ export class CreateContacts1690925872592 implements MigrationInterface {
       `INSERT INTO "BaseContact"(id, legal_name, display_name, party_id, created_at, last_updated_at, type) SELECT id, name, alias, id, created_at, last_updated_at, 'Organization' FROM "Contact"`,
     )
     await queryRunner.query(`DROP TABLE "Contact"`)
+    await queryRunner.query(
+        `CREATE TYPE "public"."partyOrigin_type_enum" AS ENUM('internal', 'external')`
+    )
+    await queryRunner.query(
+        `ALTER TABLE "PartyType" ADD COLUMN "origin" "public"."partyOrigin_type_enum" NOT NULL`
```
Comment (nklomp) [inline packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts:92]: Why are you altering a table, that is literally created in this migration file?

File: packages/data-store/src/entities/contact/PartyTypeEntity.ts [hunk 2]
```diff
@@ -14,6 +24,9 @@ export class PartyTypeEntity {
   @Column('simple-enum', { name: 'type', enum: PartyTypeEnum, nullable: false, unique: false })
   type!: PartyTypeEnum
 
+  @Column('simple-enum', { name: 'origin', enum: PartyOrigin, nullable: true, unique: false })
```
Comment (nklomp) [inline packages/data-store/src/entities/contact/PartyTypeEntity.ts:27]: I think we discussed in the sprint start to have a default (external) and make it non-nullable

File: packages/data-store/src/migrations/sqlite/1690925872693-CreateContacts.ts [hunk 3]
```diff
@@ -149,6 +149,9 @@ export class CreateContacts1690925872693 implements MigrationInterface {
       `INSERT INTO "BaseContact"(id, legal_name, display_name, party_id, created_at, last_updated_at, type) SELECT id, name, alias, id, created_at, last_updated_at, 'Organization' FROM "Contact"`,
     )
     await queryRunner.query(`DROP TABLE "Contact"`)
+    await queryRunner.query(
+        `ALTER TABLE "PartyType" ADD COLUMN "origin" varchar CHECK( "origin" IN ('internal', 'external') )`
```
Comment (nklomp) [inline packages/data-store/src/migrations/sqlite/1690925872693-CreateContacts.ts:153]: Why are you altering a table, that is literally created in this migration file?

File: packages/data-store/src/types/contact/contact.ts [hunk 4]
```diff
@@ -150,9 +150,15 @@ export type Contact = NaturalPerson | Organization
 export type NonPersistedContact = NonPersistedNaturalPerson | NonPersistedOrganization
 export type PartialContact = PartialNaturalPerson | PartialOrganization
 
+export enum PartyOrigin {
+  INTERNAL = 'internal',
```
Comment (nklomp) [inline packages/data-store/src/types/contact/contact.ts:154]: Given it is likely this enum will be used in DTOs have the casing the same!

File: packages/data-store/src/types/contact/contact.ts [hunk 5]
```diff
@@ -150,9 +150,15 @@ export type Contact = NaturalPerson | Organization
...
+  EXTERNAL = 'external',
+}
+
 export type PartyType = {
   id: string
   type: PartyTypeEnum
+  origin?: PartyOrigin
```
Comment (nklomp) [inline packages/data-store/src/types/contact/contact.ts:161]: We discussed in the sprint start to make them mandatory with a default of external

**Review by nklomp — CHANGES_REQUESTED** [review]
See remarks about what was discussed during sprint start as well as altering tables that are created in the same file

File: packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts (hunk 1 above)
Comment (maikel-maas) [reply packages/data-store/src/migrations/postgres/1690925872592-CreateContacts.ts]: Adding the column at table creation

File: packages/data-store/src/migrations/sqlite/1690925872693-CreateContacts.ts (hunk 3 above)
Comment (maikel-maas) [reply packages/data-store/src/migrations/sqlite/1690925872693-CreateContacts.ts]: Adding the column at table creation

File: packages/data-store/src/entities/contact/PartyTypeEntity.ts (hunk 2 above)
Comment (maikel-maas) [reply packages/data-store/src/entities/contact/PartyTypeEntity.ts]: Made it non-nullable with external as default

File: packages/data-store/src/types/contact/contact.ts (hunk 4 above)
Comment (maikel-maas) [reply packages/data-store/src/types/contact/contact.ts]: Made all lower case

File: packages/data-store/src/types/contact/contact.ts (hunk 5 above)
Comment (maikel-maas) [reply packages/data-store/src/types/contact/contact.ts]: Made origin mandatory in PartyType

**Review by nklomp — APPROVED** [review]
Making the enums all caps would have made more sense to me, but approving

**Pull request merged**